lightning/src/ln/channelmanager.rs

```diff
@@ -4699,6 +4731,7 @@ impl<Signer: Sign, M: Deref , T: Deref , K: Deref , F: Deref , L: Deref >
 							if let Some(short_id) = chan.get_short_channel_id() {
 								short_to_id.remove(&short_id);
 							}
+							self.pending_events.lock().unwrap().push(events::Event::ChannelClosed { channel_id: chan.channel_id(),  reason: ClosureReason::DisconnectedPeer });
```

Comment:
Given this is the case for `no_connection_possible = false`, is `DisconnectedPeer` still the appropriate reason? Seems like it'd be something else

Reply:
Good catch! I think this is still `DisconnectedPeer`, but we should update the docs to note that it can also be for "peer disconnected before we finished negotiating the channel creation"

Reply:
Alternatively, we could split the two into `PeerUnconnectable` and `PeerDisconnected`?

Reply:
Hmm, I also think `DisconnectedPeer` should be renamed to `NoConnectionPossible`, then (since DisconnectedPeer isn't in itself a valid closure reason)

Reply:
Taking the comment and adding a TODO. We have already an issue for follow-ups!